### PR TITLE
Update docker ssh passthrough docs

### PR DESCRIPTION
**Pull request by pat-s**
As discussed in Discord with @zeripath, a major rewrite and simplification of the docker SSH passthrough docs.

## Changelog summary

- Remove the option to create a softlink for `.ssh/authorized_keys` → confusing and unstable due to `SSH_AUTHORIZED_KEYS_BACKUP` option
- Explain in more details how the SSH request process is happening from start to end
- Pretty much rewrite the section from scratch in passive voice

## Related issues

- [https://github.com/go-gitea/gitea/issues/12625](https://github.com/go-gitea/gitea/issues/12625)
- [https://github.com/go-gitea/gitea/issues/13716](https://github.com/go-gitea/gitea/issues/13716)
- [https://github.com/go-gitea/gitea/issues/12625](https://github.com/go-gitea/gitea/issues/12625)
- [https://github.com/go-gitea/gitea/issues/11318](https://github.com/go-gitea/gitea/issues/11318)
- and possibly more

---

I am not sure if PR https://github.com/go-gitea/gitea/pull/12626 is valid or not - for me it worked with the current setup on the latest dev version. But this is not necessarliy important for this PR.

### CODE REVIEW

File: docs/content/doc/installation/with-docker.en-us.md [hunk 1]
```diff
@@ -301,88 +302,66 @@ docker-compose up -d
...
-Now, proceed with one of the points given below:
-
-- symlink the container `.ssh/authorized_keys` file to your git user `.ssh/authorized_keys`.
-This can be done on the host as the `/var/lib/gitea` directory is mounted inside the container under `/data`:
+In the next step a file named `/app/gitea/gitea` (with executable permissions) needs to be created on the host. This file will issue the SSH forwarding from the host to the container. Add the following contents to `/app/gitea/gitea`:
 
-```
-ln -s /var/lib/gitea/git/.ssh/authorized_keys /home/git/.ssh/authorized_keys
+```bash
+ssh -p 2222 -o StrictHostKeyChecking=no [git@127.0.0.1](mailto:git@127.0.0.1) "SSH_ORIGINAL_COMMAND=\"$SSH_ORIGINAL_COMMAND\" $0 $@"
```
Comment (techknowlogick) [inline docs/content/doc/installation/with-docker.en-us.md]: I think this mailto may have snuck in unintentionally.

**Comment by TwistTheNeil** [comment]
Thanks for this PR, @pat-s, it helped me get ssh passthrough working :)

File: docs/content/doc/installation/with-docker.en-us.md (hunk 1 above)
Comment (pat-s) [reply docs/content/doc/installation/with-docker.en-us.md]: Feel free to remove it - I just used what Gitea already uses in this place.

Comment (pat-s) [reply docs/content/doc/installation/with-docker.en-us.md]: Removed it.

**Review by techknowlogick — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Pull request merged**